src/Controls/src/Core/Xaml/TypeConversionExtensions.cs

```diff
@@ -71,42 +72,51 @@ static class TypeConversionExtensions
 		{
 			Func<TypeConverter> getConverter = () =>
 			{
-				MemberInfo memberInfo;
+				if (minfoRetriever != null && minfoRetriever() is MemberInfo memberInfo)
+				{
+					if (memberInfo.CustomAttributes.GetTypeConverterType() is Type converterType)
```

Comment:
I'd add a second cache level here with a ConcurrentDictionary<MemberInfo, TypeConverter> with the double benefit of not having to cycle through CustomAttributes and avoid re-creating those converters.

Also, we should cache the fact that no Converter has been found for that MemberInfo to fallback faster, and avoid iterating through the CustomAttributes

Reply:
As mentioned in a different comment, these 2 caches should be generated at build time using a source generator, but this isn't a blocker for this PR